### ISSUE TITLE
Execute the deployopenshift.sh script and report an error

### ISSUE THREAD

**Issue by 963029755** [Microsoft/openshift-container-platform#153]
**Describe the bug**
An error occurred in azure executing your script in China
Modify the script link: https://openshift1.blob.core.chinacloudapi.cn/script/deployOpenShift.sh

**To Reproduce**
An error was reported after running the script

**Expected behavior**
Successfully deployed openshift automation scripts in azure China

**Screenshots**
![error](https://user-images.githubusercontent.com/46158660/52771020-b5233f80-306f-11e9-8d5f-f477c9c6fd13.jpg)

**stdout**
Include the last 100 lines of stdout from Bastion host - see troubleshooting https://docs.microsoft.com/en-us/azure/virtual-machines/linux/openshift-troubleshooting

**Template Information (please complete the following information):**
 - Branch: [e.g. master]

**Additional context**

**Comment by haroldwongms** [comment]
Are you sure you have the private key configured correctly?  No passphrase assigned?  Did you create the Key Vault secret properly?

**Comment by 963029755** [comment]
All of these were created successfully, with the creation process on azure:
1. Create resource groups
2. Create a keystore
3. Create the key
4. Assign permissions
5. Upload the certificate/public key
Create the run script
Do you think the script is modified correctly? The virtual machine "-0" could not be found because of an error.

**Comment by haroldwongms** [comment]
Did you fork the templates and then modify them prior to deploying?

**Comment by 963029755** [comment]
I changed your template azuredeploy json, azuredeploy. The parameters. The json, deployOpenShift. Sh these three document link is as follows, please you check the problem:
Azuredeploy. Json link: https://openshift1.blob.core.chinacloudapi.cn/script/azuredeploy.json
Azuredeploy. The parameters. The json link: https://openshift1.blob.core.chinacloudapi.cn/script/azuredeploy.parameters.json
DeployOpenShift. Sh link: https://openshift1.blob.core.chinacloudapi.cn/script/deployOpenShift.sh

**Comment by 963029755** [comment]
Is there anything running Ansible Inventory Ansible Inventory in the script?

**Comment by 963029755** [comment]
I found myself deploying as if the parameters weren't accurate...Where was the master parameter not given correctly in the script? Please give me your opinion.

**Comment by haroldwongms** [comment]
Sorry for the delay.  I have been working on making a major update to the templates and scripts.  I have already incorporated the fixex into the latest version of the templates and have opened a PR with a major revision.  Once this gets reviewed and we merge, all should be good.

**Comment by 963029755** [comment]
The picture in the attachment is an error message. I'd like to trouble you to have a look at it. Thank you very much.

![2019-3-11](https://user-images.githubusercontent.com/46158660/54099167-d4a15400-43f2-11e9-8a22-5e5356a4a099.png)

cluster-bastion主机上的文件stderr  /var/lib/waagent/custom-script/download/1/stderr
![2019-03-11](https://user-images.githubusercontent.com/46158660/54099088-855b2380-43f2-11e9-9da3-6231a21ad465.jpg)

**Comment by haroldwongms** [comment]
I just realized that this is an issue with deployment in China (21ViaNet).  I have not ever tested this in China as I don't have access.

**Comment by 963029755** [comment]
Yes, if deployed on azure in China, there will be an error and no privilege to execute, because the current user on azure is not root, will it be related to this? Thank you...

**Comment by haroldwongms** [comment]
Can you try the latest in master branch and let me know what happens?  Better to troubleshoot the current release that I know works in Azure Public today.

**Comment by 963029755** [comment]
Can you come up with a script for the deployment of azure in China? We can provide resources and labor costs here, and we can also come up with it.

**Comment by haroldwongms** [comment]
what error are you getting with the updated scripts?  I don't have the ability to deploy to Azure China so I have no idea how to test.

**Comment by 963029755** [comment]
The latest deployment has been tried, and we will provide you with resources for Azure China account, which is already a headache. Thank you for your reply.

**Comment by 963029755** [comment]
![1554299397(1)](https://user-images.githubusercontent.com/46158660/55484088-7155bb00-565a-11e9-9ae2-26745241e9bd.jpg)
![Uploading 0219-0403.png…]()
Sorry,  run the command to create the resource:

az group deployment create --name ocpdeployment --template-file azuredeploy.json --parameters @azuredeploy.parameters.json --resource-group openshift-cluster

Then the execution is wrong... How did you deploy it?

**Comment by haroldwongms** [comment]
Will address in templates once I confirm an available supported RHEL image in Azure China.

**Comment by 963029755** [comment]
Good morning:


        Well, thank you for uploading it to azure China from our company. I'm putting it on the web these days.
"marketplaceOsImage": {
            "value": {
                "publisher": "yungoalbj",
                "offer": "redhat75",
                "sku": "standard",
                "version": "1.0.0"
            }
        },


        Can you write the "aadClientSecret","openshiftPassword","rhsmPasswordOrActivationKey","sshPrivateKey" keystore into your script? I'm trying to write the template script of the keystore into your script these days, but I always make mistakes. Can you help me write this into your script?


        Thank you!
        I wish you every success in your work.



------------------ 原始邮件 ------------------
发件人: "Harold Wong"<notifications@github.com>;
发送时间: 2019年4月24日(星期三) 上午8:55
收件人: "Microsoft/openshift-container-platform"<openshift-container-platform@noreply.github.com>;
抄送: "劉刚"<963029755@qq.com>; "Author"<author@noreply.github.com>; 
主题: Re: [Microsoft/openshift-container-platform] Execute thedeployopenshift.sh script and report an error (#153)




Will address in templates once I confirm an available supported RHEL image in Azure China.
